### PR TITLE
Backend Model and Controller

### PR DESCRIPTION
**Pull request by gvaradarajan**
All instructions for getting the server up and running are in the README in the server directory. To ensure that all proper technologies are present (and assuming you have pypy), you need to execute the command `pip install -r requirements.txt` before attempting to start the server. Any other Python-based technologies should be added to the requirements file. I'll implement the model for matches when Mike gets his part in. Also I'll spend some time later refactoring the code to match the style guide.

### CODE REVIEW

**Comment by MWGitHub** [comment]
For the .gitignore file can you replace all the .pyc ignores with *.pyc?

For the dictionary objects the style guide indents like [this](https://google.github.io/styleguide/pyguide.html?showone=Indentation#Indentation).

Everything else looks good except for loading the engine. It may be better to load it as a singleton and also have a config file or env var for the psql url.

File: schema.md [hunk 1]
```diff
@@ -23,10 +23,9 @@ name             | string    | not null
...
-top              | integer   | not null
-mid              | integer   | not null
-jungle           | integer   | not null
-marksman         | integer   | not null
-support          | integer   | not null
-
+role             | string    | must be between 00000 and 11111
 image            | img       | not null
+
+NB: `role` above is a binary string with digits ordered by top-mid-jungle-marksman-support,
```
Comment (MWGitHub) [inline schema.md]: Role will be easier to check against if it's an integer within the range 0b11111 to 0b00000 with bitwise operators.

**Comment by gvaradarajan** [comment]
I can't quite figure out how to use the environmental variables, make the engine a singleton, and have them play nicely with the server. I'm currently fixing everything else. Once I've done so, could we merge the pull request until I can figure out how to accomplish the rest?

**Comment by MWGitHub** [comment]
Sounds good

**Pull request merged**